src/averageAges.js

```diff
@@ -1,61 +1,52 @@
 'use strict';
 
-/**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
- * @param {object[]} people
- * @param {number} century - optional
- *
- * @return {number}
- */
+function arrayAverage(arr) {
+  const fullSum = arr.reduce((sum, a) => sum + a, 0);
+
+  return fullSum / arr.length;
+}
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter(person => person.sex === 'm');
+  const menOfTheCentury = !century ? men
+    : men.filter(person => Math.ceil(person.died / 100) === century);
```

Comment:
could you try adding the condition directly to the men declaration?
this will help you avoid another filter

Reply:
and also move branching of ternary operathor to the new line